integration-tests/actions/seth/refund.go

```diff
@@ -262,94 +262,108 @@ func ReturnFunds(log zerolog.Logger, sethClient *seth.Client, chainlinkNodes []c
 				return err
 			}
 
-			publicKey := decryptedKey.PrivateKey.Public()
-			publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
-			if !ok {
-				return errors.New("error casting public key to ECDSA")
-			}
-			fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
-
-			balance, err := sethClient.Client.BalanceAt(context.Background(), fromAddress, nil)
+			err = sendAllFundsIfPossible(log, client, decryptedKey.PrivateKey)
 			if err != nil {
-				return err
+				publicKey := decryptedKey.PrivateKey.Public()
+				publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+				if !ok {
+					return errors.New("error casting public key to ECDSA")
+				}
+				failedReturns = append(failedReturns, crypto.PubkeyToAddress(*publicKeyECDSA))
 			}
+		}
+	}
 
-			if balance.Cmp(big.NewInt(0)) == 0 {
-				log.Info().
-					Str("Address", fromAddress.String()).
-					Msg("No balance to return. Skipping return.")
-			}
+	if len(failedReturns) > 0 {
+		return fmt.Errorf("failed to return funds from Chainlink nodes to default network wallet for addresses: %v", failedReturns)
+	}
 
-			// if not set, it will be just set to empty string, which is okay as long as gas estimation is disabled
-			txPriority := sethClient.Cfg.Network.GasPriceEstimationTxPriority
-			txTimeout := sethClient.Cfg.Network.TxnTimeout.Duration()
+	log.Info().Msg("Successfully returned funds from all Chainlink nodes to default network wallets")
 
-			if sethClient.Cfg.IsExperimentEnabled(seth.Experiment_SlowFundsReturn) {
-				txPriority = "slow"
-				thirtyMinutes := time.Duration(30 * time.Minute)
-				txTimeout = thirtyMinutes
-			}
+	return nil
+}
 
-			estimations := sethClient.CalculateGasEstimations(seth.GasEstimationRequest{
-				GasEstimationEnabled: sethClient.Cfg.Network.GasPriceEstimationEnabled,
-				FallbackGasPrice:     sethClient.Cfg.Network.GasPrice,
-				FallbackGasFeeCap:    sethClient.Cfg.Network.GasFeeCap,
-				FallbackGasTipCap:    sethClient.Cfg.Network.GasTipCap,
-				Priority:             txPriority,
-			})
-
-			var maxTotalGasCost *big.Int
-			if sethClient.Cfg.Network.EIP1559DynamicFees {
-				maxTotalGasCost = new(big.Int).Mul(big.NewInt(0).SetInt64(sethClient.Cfg.Network.TransferGasFee), estimations.GasFeeCap)
-			} else {
-				maxTotalGasCost = new(big.Int).Mul(big.NewInt(0).SetInt64(sethClient.Cfg.Network.TransferGasFee), estimations.GasPrice)
-			}
+func sendAllFundsIfPossible(log zerolog.Logger, sethClient *seth.Client, fromPrivateKey *ecdsa.PrivateKey) error {
+	publicKey := fromPrivateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	if !ok {
+		return errors.New("error casting public key to ECDSA")
+	}
 
-			toSend := new(big.Int).Sub(balance, maxTotalGasCost)
+	fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+	balance, err := sethClient.Client.BalanceAt(context.Background(), fromAddress, nil)
```

Comment:
it should use the context provided from the test

Reply:
will fix